bukkit/src/main/java/me/lokka30/treasury/plugin/bukkit/MessagesImpl.java

```diff
@@ -0,0 +1,56 @@
+package me.lokka30.treasury.plugin.bukkit;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.util.Collection;
+import java.util.function.Supplier;
+import me.lokka30.treasury.plugin.core.config.messaging.MessageKey;
+import me.lokka30.treasury.plugin.core.config.messaging.Messages;
+import me.lokka30.treasury.plugin.core.config.messaging.MessagesConfigAccessor;
+import org.bukkit.configuration.file.YamlConfiguration;
+import org.jetbrains.annotations.NotNull;
+
+public class MessagesImpl extends Messages {
+
+    private final Treasury plugin;
+
+    public MessagesImpl(Treasury plugin) {
+        super(supplyMessages(plugin).get());
+        this.plugin = plugin;
+    }
+
+    private static Supplier<MessagesConfigAccessor> supplyMessages(Treasury plugin) {
+        return () -> {
+            File file = new File(plugin.getDataFolder(), "messages.yml");
+            if (!file.exists()) {
+                if (!file.getParentFile().exists()) {
+                    file.getParentFile().mkdirs();
+                }
+                try (InputStream in = plugin.getClass().getClassLoader().getResourceAsStream("messages.yml")) {
+                    Files.copy(in, file.getAbsoluteFile().toPath());
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+                return MessagesConfigAccessor.EMPTY;
+            }
+            YamlConfiguration config = YamlConfiguration.loadConfiguration(file);
+            return (MessagesConfigAccessor) config::get;
+        };
+    }
+
+    @Override
+    public void generateMissingEntries(@NotNull Collection<MessageKey> keys) {
+        // todo: find another way!!!
```

Comment:
[Configurate](https://github.com/SpongePowered/Configurate) can do this, with comment support :)

EDIT: Am I seeing this wrong or does your current implementation just delete all user-entered messages if the missing entries are generated?

Reply:
The current implementation regenerates the file if it finds missing entries. That's why I added todo to find another way. I'm also thinking of [AnnotationConfig](https://github.com/MrIvanPlays/AnnotationConfig) which is able to handle missing options and config generation with comments all by itself. It needs more discussion.

Reply:
Personally would prefer something dependent on [SnakeYaml Engine](https://bitbucket.org/asomov/snakeyaml-engine/src/master/) (YAML 1.2 spec, built off SnakeYaml), but I haven't yet found a library that does that satisfactorily. Looks like Configurate plans to update eventually, though the issue for it has been pushed back twice. That may also solve the "yes" and "no" issues in paths, I believe a lot of SnakeYaml's original quirks were removed.